gravitee-am-identityprovider/gravitee-am-identityprovider-api/src/main/java/io/gravitee/am/identityprovider/api/social/SocialAuthenticationProvider.java

```diff
@@ -36,20 +40,23 @@ public interface SocialAuthenticationProvider extends AuthenticationProvider {
     @Deprecated
     Request signInUrl(String redirectUri, String state);
 
+
     /**
      * Generate the signIn Url in asynchronous way
-     * to avoid blocking thread when an http
-     * call out is required to generate the url.
-     *
-     * @param redirectUri
-     * @return
+     * Allows introspecting and modifying the state JWT; the state MUST be encoded using the provided stateEncoder
      */
+    default Maybe<Request> asyncSignInUrl(String redirectUri, JWT state, Function<JWT, Single<String>> stateEncoder) {
```

Comment:
question: should we create an interface/function interface named StateEncoder instead of providing the Function with Generic :thinking: I wondering if an interface could simplify future evolution :shrug:

Reply:
added in the management PR